src/AssetRegistry/MainRegistry.sol

```diff
@@ -171,67 +176,57 @@ contract MainRegistry is Ownable {
         return true;
     }
 
-    /* ///////////////////////////////////////////////////////////////
-                        WHITE LIST LOGIC
-    /////////////////////////////////////////////////////////////// */
-
     /**
-     * @notice Checks for a list of tokens and a list of corresponding IDs if all tokens are white-listed
-     * @param assetAddresses The list of token addresses that needs to be checked
-     * @param assetIds The list of corresponding token Ids that needs to be checked
-     * @dev For each token address, a corresponding id at the same index should be present,
-     * for tokens without Id (ERC20 for instance), the Id should be set to 0
-     * @return A boolean, indicating of all assets passed as input are whitelisted
+     * @notice Batch process multiple assets
+     * @param assetAddresses An array of addresses of the assets
+     * @param assetIds An array of asset ids
+     * @param amounts An array of amounts to be deposited
+     * @dev processDeposit in the pricing module checks whehter
+     *    it's allowlisted and updates the maxExposure
      */
-    function batchIsWhiteListed(address[] calldata assetAddresses, uint256[] calldata assetIds)
-        public
-        view
-        returns (bool)
-    {
+    function batchProcessDeposit(
+        address[] calldata assetAddresses,
+        uint256[] calldata assetIds,
+        uint256[] calldata amounts
+    ) public onlyVault {
         uint256 addressesLength = assetAddresses.length;
-        require(addressesLength == assetIds.length, "LENGTH_MISMATCH");
+        require(addressesLength == assetIds.length && addressesLength == amounts.length, "MR_BPD: LENGTH_MISMATCH");
 
         address assetAddress;
         for (uint256 i; i < addressesLength;) {
             assetAddress = assetAddresses[i];
             if (!inMainRegistry[assetAddress]) {
```

Comment:
Why not:
require(inMainRegistry[assetAddress], "MR_BPD: Asset not in mainreg");
And remove if and else.
Also do we need the check, we don't do it in batchProcessWithdrawal?

Reply:
I'll update the require, thinking about whether or not that inMainReg check is needed...

Reply:
Could be removed, #169